File: app/src/main/java/com/aso/senote/data/source/local/converter/DateConverter.java
/*
 * Copyright (c) ASO 2020.
 */

package com.aso.senote.data.source.local.converter;

import androidx.room.TypeConverter;

import java.util.Calendar;

public class DateConverter {
    @TypeConverter
    public static Long calenderToDateStamp(Calendar calendar) {
        return calendar.getTimeInMillis();
    }

    @TypeConverter
    public static Calendar dateStampToCalender(long value) {
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(value);
        return calendar;
    }
}

File: app/src/main/java/com/aso/senote/data/source/local/converter/DateConverter.kt
/*
 * Copyright (c) ASO 2020.
 */
package com.aso.senote.data.source.local.converter

import androidx.room.TypeConverter
import java.util.*

object DateConverter {
    @JvmStatic
    @TypeConverter
    fun calenderToDateStamp(calendar: Calendar): Long {
        return calendar.timeInMillis
    }

    @JvmStatic
    @TypeConverter
    fun dateStampToCalender(value: Long): Calendar {
        val calendar = Calendar.getInstance()
        calendar.timeInMillis = value
        return calendar
    }
}